src/server/main_service.cc

```diff
@@ -967,7 +967,7 @@ optional<ErrorReply> Service::CheckKeysOwnership(const CommandId* cid, CmdArgLis
 // Return OK if all keys are allowed to be accessed: either declared in EVAL or
 // transaction is running in global or non-atomic mode.
 OpStatus CheckKeysDeclared(const ConnectionState::ScriptInfo& eval_info, const CommandId* cid,
```

Comment:
maybe this function can return ErrorReply? instead of status and out param in case of error

Reply:
This is a great idea! Done.